ors-api/src/main/java/org/heigit/ors/api/responses/export/topojson/TopoJsonExportResponse.java

```diff
@@ -0,0 +1,134 @@
+package org.heigit.ors.api.responses.export.topojson;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import lombok.Builder;
+import lombok.Getter;
+import org.heigit.ors.common.Pair;
+import org.heigit.ors.export.ExportResult;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.LineString;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.heigit.ors.export.ExportResult.TopoGeometry;
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@JsonPropertyOrder({"type", "transform", "objects", "arcs", "bbox"})
+@Getter
+@Builder
+public class TopoJsonExportResponse implements Serializable {
+
+    @JsonProperty(value = "type")
+    @Builder.Default
+    private String type = "Topology";
+    @JsonProperty("objects")
+    @Builder.Default
+    private Network objects = null;
+    @JsonProperty("arcs")
+    @Builder.Default
+    private List<Arc> arcs = new LinkedList<>();
+    @JsonProperty("bbox")
+    @Builder.Default
+    private List<Double> bbox = new ArrayList<>(4);
+
+    public static TopoJsonExportResponse fromExportResult(ExportResult exportResult) {
+        BBox bbox = new BBox();
+        LinkedList<Geometry> geometries = new LinkedList<>();
+        LinkedList<Arc> arcsLocal = new LinkedList<>();
+        int arcCount = 0;
+        if (exportResult.getTopoGeometries().isEmpty()) {
+            // If OSM ids are not present, we are creating a simple topology with geometries for every edge
```

Comment:
I think this comment expresses a good opportunity for refactoring for readability:
- `ExportResult` could have a method `boolean hasOsmIds()` called in the if clause (switching if and else branches)
- The bodies of if and else could be extracted as methods returning the geometries, e.g., `createSimpleTopologyWithGeometriesPerEdge(...)` and `create<WhateverFitsHere>Topology...(...)`

Reply:
adressed